bundles/org.eclipse.core.commands/src/org/eclipse/core/internal/commands/util/Util.java

```diff
@@ -255,10 +255,11 @@ public static final String getHelpContextId(Command command) {
 
 		String contextId = null;
 		if (method != null) {
-			boolean accessible = method.isAccessible();
-			method.setAccessible(true);
+			boolean accessible = method.canAccess(command);
 			try {
-				contextId = (String) method.invoke(command);
+				if (method.trySetAccessible()) {
+					contextId = (String) method.invoke(command);
+				}
 			} catch (Exception e) {
 				// do nothing
 			}
```

Comment:
wouldn't the finally block also need be adjusted?

Reply:
Also: `Command#getHelpContextId()` says clients shally use `CommandManager#getHelpContextId(Command)` instead. Can't we call that here?

Reply:
there is no final block here, `CommandManager does a lot more then just calling the method. I don't want to break anything.

Reply:
ah, right, there is no finally block here. but the same line of thought applies:

The condition is changed from `isAccessible()` to `canAccess()`, and now the defensive `if(tryAccess())` is used, but in line 266 `setAccessible()` is still called unconditionally which can then still fail.

I propose to also move that 'undo/cleanup' line inside the `if(try)` block, since otherwise there is nothing to undo and it could needlessly raise an exception.

Reply:
OK